### PR TITLE
MOTECH-2203 Added filters for Boolean values in tasks module

### PR DESCRIPTION
**Pull request by shanmukhm**

### CODE REVIEW

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

File: modules/tasks/tasks/src/main/resources/webapp/messages/messages.properties [hunk 1]
```diff
@@ -348,7 +348,10 @@ task.validation.error.provider.name=Invalid data provider name: ''{0}''. Provide
 task.string=String
 task.number=Number
 task.date=Date
+task.boolean=Boolean
 
+task.AND=And
+task.OR=Or
```
Comment (pgesek) [inline modules/tasks/tasks/src/main/resources/webapp/messages/messages.properties]: why are these all capital letters? make them consistent with other options

File: modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/util/TaskFilterExecutor.java [hunk 2]
```diff
@@ -185,6 +187,30 @@ private boolean checkFilterForNumber(Filter filter, BigDecimal param) {
...
 
+    private boolean checkFilterForBoolean(Filter filter, Boolean param) {
+        OperatorType operatorType = OperatorType.fromString(filter.getOperator());
+        boolean result = false;
+        boolean expressionValue = Boolean.parseBoolean(filter.getExpression());
+
+        if (operatorType != null) {
+            switch (operatorType) {
+                case EXIST:
+                    result = true;
```
Comment (pgesek) [inline modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/util/TaskFilterExecutor.java]: this should we check if the param is true? (result = param) If it always returns true it's pointless

File: modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/util/TaskFilterExecutor.java [hunk 3]
```diff
@@ -185,6 +187,30 @@ private boolean checkFilterForNumber(Filter filter, BigDecimal param) {
         return result;
     }
 
+    private boolean checkFilterForBoolean(Filter filter, Boolean param) {
```
Comment (pgesek) [inline modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/util/TaskFilterExecutor.java]: extend the unit tests to test these boolean filters

**Comment by pgesek** [comment]
test this please

**Comment by pgesek** [comment]
test this please

**Comment by pgesek** [comment]
test this please

**Comment by pgesek** [comment]
test this please

**Pull request merged**